crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -899,17 +908,21 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientStatePtr {
                 surface_y,
                 ..
             } => {
-                state.serial = serial;
```

Comment:
I believe we might need this serial for receiving data offers for file drag and drop?

Reply:
Drag and drop uses the serial received at `wl_data_device::Event::Enter` so it still works :slightly_smiling_face:

Reply:
To add to this, I think we should add a `Serial` wrapper kind of like Chromium has, since different serial types are used for different things.

Reply:
Agreed, thanks for clarifying!